From cfa57e264efa138e99257fbfbe18449150a96c4d Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Thu, 14 Feb 2019 20:18:04 +0100
Subject: [PATCH] install: string_hash_func should not be fed with NULL

If kmod_module_get_path returns NULL, we should skip that entry,
the hash function does not like NULL pointers.

(cherry picked from commit fc141f22869bad2e5409d1cc555c1a42ea738343)

Resolves: #1868525
---
 install/dracut-install.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/install/dracut-install.c b/install/dracut-install.c
index 51f79422..f8ec9a59 100644
--- a/install/dracut-install.c
+++ b/install/dracut-install.c
@@ -1248,6 +1248,9 @@ static int install_dependent_modules(struct kmod_list *modlist)
                 mod = kmod_module_get_module(itr);
                 path = kmod_module_get_path(mod);
 
+                if (path == NULL)
+                        continue;
+
 		if (check_hashmap(items_failed, path))
 			return -1;
 
@@ -1257,7 +1260,7 @@ static int install_dependent_modules(struct kmod_list *modlist)
 
                 name = kmod_module_get_name(mod);
 
-                if ((path == NULL) || (arg_mod_filter_noname && (regexec(&mod_filter_noname, name, 0, NULL, 0) == 0))) {
+                if (arg_mod_filter_noname && (regexec(&mod_filter_noname, name, 0, NULL, 0) == 0)) {
                         continue;
                 }
 

